src/ddscxx/src/org/eclipse/cyclonedds/topic/qos/TopicQosDelegate.cpp

```diff
@@ -150,6 +157,7 @@ TopicQosDelegate::ddsc_qos() const
     priority_    .delegate().set_c_policy(qos);
     lifespan_    .delegate().set_c_policy(qos);
     ownership_   .delegate().set_c_policy(qos);
+    datarepresentation_.delegate().set_c_policy(qos);
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro. (BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).

Reply:
Will do.